### PR TITLE
Switch to GHA for CI

### PR DESCRIPTION
**Pull request by kingthorin**
Drop travis which is currently broken.

@michaelficarra @bakkot you can see test runs here: 
- push
    - https://github.com/kingthorin/saltest/actions/runs/1671719996
- PR
    - https://github.com/kingthorin/saltest/pull/1
        - https://github.com/kingthorin/saltest/actions/runs/1671838297

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
I guess you'll need to adjust the branch protection rules to merge this and then to enforce this.

**Comment by bakkot** [comment]
Thanks! I've been backlogged on other stuff for a while but am hoping to get back to this project within the next month or so, and I'll review this then.

**Comment by kingthorin** [comment]
Heya, no problem. I know you've both been busy and fixing travis is on the todo list. I just figured if I could get this done then perhaps it would make life easier moving on with other stuff.

I just pushed a fix for the CLA check, that allows it to remain compatible with the existing script which assumes use of travis.

**Comment by kingthorin** [comment]
Hey crew, any news on this? It would be great to keep moving this forward. I'd like to get that typo PR in, submit some SAST fixes, and maybe tackle the trusted types changes (if I can figure out how/where to fit those in).

**Comment by bakkot** [comment]
This is now the next-to-top item on my stack, so I hope to be getting back to this within a week or two.

**Comment by kingthorin** [comment]
Hey crew, how are things looking?

**Comment by kingthorin** [comment]
@bakkot whats the word?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      - name: CLA Check
+        if: github.event_name == 'pull_request'
+        env:
+          TRAVIS_COMMIT_RANGE: ${{ github.event.pull_request.base.sha }}...${{ github.event.pull_request.head.sha }}
+        run: curl https://raw.githubusercontent.com/shapesecurity/CLA/HEAD/cla-travis.sh | bash
+      - name: Setup Java
+        if: success()
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
```
Comment (bakkot) [inline .github/workflows/ci.yml:34]: Any particular reason to use adopt instead of whatever the default is?

**Review by bakkot — APPROVED** [review]
Works for me.

**Comment by bakkot** [comment]
Sorry for the delay! I have _finally_ finished the other stuff I was working on, and will have more time to catch up OSS work for a while.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kingthorin) [reply .github/workflows/ci.yml:34]: Version and distributed are mandatory, there is no default.

I have experience with adopt and seems widely used in the industry.

**Comment by kingthorin** [comment]
Thanks

**Comment by bakkot** [comment]
(CI failed but will be fixed by https://github.com/shapesecurity/salvation/pull/252.)

**Comment by kingthorin** [comment]
Thanks!

**Pull request merged**